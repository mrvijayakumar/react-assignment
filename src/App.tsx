import React, {useState, useEffect} from 'react';
import Grid from './components/grid';
import Modal from './components/modal';
import {FetchCountries} from './api';
import LoadingIndicator from './components/preloader';
import {Labels} from './languages';
import  './style.css';
function App() {   
  const countriesList = FetchCountries('/all');
  const [gridData, setGridData] = useState([]);
  const [isModalOpen, setModalOpen] = useState(false);
  const [modalData, setModalData] = useState({});
  const gridPreference = {
    columndef : [
      {'id' : 'name', 'name' : Labels('country')},      
      {'id' : 'capital', 'name' : Labels('capital')},
      {'id' : 'population', 'name' : Labels('population')},
      {'id' : 'view', 'name' : Labels('view'), 'formatter' : true}
    ],
    gridData : gridData,
    gridUniqueId : "numericCode"
  }; 
  const viewFormatter = (id : any) => {
    const dataFormodal = gridData.filter((datum :any)=> { return datum.numericCode === id;})[0];
    setModalData(dataFormodal);
    setModalOpen(id);
  }  
  useEffect(() => {
    setGridData(countriesList.data);
  }, [countriesList]); 
  return <>
    {!countriesList.dataLoaded && <LoadingIndicator /> }
    {countriesList.dataLoaded && 
    <div className="container">
      <div className="column">
        <Grid preference={gridPreference} formatter={viewFormatter} />
      </div>
    </div>   
    }
    <Modal open={isModalOpen} data={modalData} />
  </>;
}
export default App;
