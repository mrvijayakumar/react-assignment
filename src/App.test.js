import '@testing-library/jest-dom'
import React from 'react'
import {render, fireEvent, screen} from '@testing-library/react'
import App from './App';


test('render correctly' , () => {
    const { asFragment } = render(<App />);
    expect(asFragment( <App />)).toMatchSnapshot();
});