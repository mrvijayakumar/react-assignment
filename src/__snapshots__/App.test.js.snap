// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render correctly 1`] = `
<DocumentFragment>
  <div
    class="loading"
  >
    <div
      class="loading-content"
    >
      Loading... please wait...!
    </div>
  </div>
  <div
    class="modal "
  >
    <div
      class="modal-content"
    >
      <span
        class="close-button"
      >
        Ã—
      </span>
      <p>
        <strong>
          Country Name :
        </strong>
         undefined
      </p>
      <p>
        <strong>
          Capital :
        </strong>
         undefined
      </p>
      <p>
        <strong>
          Population :
        </strong>
         undefined
      </p>
      <p>
        <strong>
          Language :
        </strong>
         
      </p>
      <p>
        <strong>
          Currency :
        </strong>
         
      </p>
      <p
        style="text-align: center;"
      >
        <button
          class="btn"
          type="button"
        >
          Ok
        </button>
      </p>
    </div>
  </div>
</DocumentFragment>
`;
